1_preprocessing_data/PyBaSiC_Pipelines/Illumination_Correction.py

```diff
@@ -29,49 +28,61 @@
-    image_files = []
-    for image_file in full_channel_path.iterdir():
-        image_file = str(image_file)
-        if image_file.endswith(file_extension):
-            image_files.append(image_file.strip(file_extension))
+    image_files = [] # List that holds all of the paths to the images in the directory for a specific channel
+    image_names = [] # List that holds the names (str) for the images that is used for saving 
+    images = [] # List of numpy arrays for the images that are read in
+
+    # This for loop will run through the specified directory, find images for a specific channel (channel name is in the file name metadata), save paths to a list, and then save the names 
+    # to a list after stripping the file_extension
+    for image_path in channels_path.iterdir():
+        image_path = str(image_path)
+        if channel in image_path:
+            image_files.append(image_path)
+            if image_path.endswith(file_extension):
+                image_names.append(image_path.strip(file_extension))
    
+    # Sorts the file paths and names into alphabetical order (puts well C at the start)
     image_files.sort()
-    channel_images = pybasic.tools.load_data(full_channel_path, file_extension, verbosity = True)
+    image_names.sort()
+
+    # This for loop will run through the paths to the images for the specified channel and load in the images to be used to illumination correction. This code was sampled from the `load_data`
+    # function in PyBaSiC
+    for i, image_file in enumerate(image_files):
+        if verbosity and (i % 10 == 0):
+            print(i, '/', len(image_files))
+        images.append(skimage.io.imread(image_file))
 
-    return channel_images, image_files
+    return images, image_names
 
 
-def run_illum_correct(channel_path: pathlib.Path, channel: str, save_path: pathlib.Path, save_channel: str, output_calc: str = 'False'):
-    """ calculates flatfield, darkfield, performs illumination correction on channel images., coverts to 8-bit and saves images into designated folder
+def run_illum_correct(channels_path: pathlib.Path, save_path: pathlib.Path, channel: str, output_calc: str = 'False'):
+    """ calculates flatfield, darkfield, performs illumination correction on channel images, coverts to 8-bit and saves images into designated folder
 
     Parameters:
-        channel_path (pathlib.Path): path to directory where image channels are located
+        channels_path (pathlib.Path): path to directory where the all images for each channel are stored
         channel (str): name of channel
-        save_path (pathlib.Path): path to directory where the corrected images will be saved to designated folders
-        channel (str): name of channel with "_Corrected" (or however you want to designate the images) to create folder to save to
+        save_path (pathlib.Path): path to directory where the corrected images will be saved to
```

Comment:
Can you add details describing what `output_calc` is? I am not seeing it here (sorry if I'm missing something obvious!)

Reply:
Oh, I see you have it described in the notebook:

> f you want to output the flatfield and darkfield calculations, then put "output_calc=True". If not, leave out from function since it is defa

Can you add some of the detail in the function?

Also, where you add this note in the `.ipynb` file is very looooong. Can you break this up into a multi-line comment?